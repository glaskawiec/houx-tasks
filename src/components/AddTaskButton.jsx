import React from 'react';
import { Button } from 'react-bootstrap';
import { useHoux } from 'houx';
import { openAddModal } from '../flux/actions/tasksActions';

const AddTaskButton = () => {
  const { dispatch } = useHoux();

  const onAddTaskButtonClick = () => {
    dispatch(openAddModal());
  };

  return (
    <Button data-testid="add-task-button" onClick={onAddTaskButtonClick}>Add Task</Button>
  );
};

export default AddTaskButton;
