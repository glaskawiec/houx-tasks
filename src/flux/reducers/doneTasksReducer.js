import { DONE_TASKS_ADD, DONE_TASKS_REMOVE } from "../actionTypes/doneTasksActionTypes";

export const initialState = [];

const doneTasksReducer = (state = initialState, action) => {
    switch (action && action.type) {
        case DONE_TASKS_ADD:
            return [...state, action.task]
        case DONE_TASKS_REMOVE:
            return state.filter((task) => task.id !== action.id)
        default: return state;
    }
}

export default doneTasksReducer;