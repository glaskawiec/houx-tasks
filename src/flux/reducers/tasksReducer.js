import {
  TASKS_ADD,
  TASKS_REMOVE,
  TASKS_COMPLETE,
  TASKS_EDIT,
  TASKS_OPEN_ADD_MODAL,
  TASKS_OPEN_EDIT_MODAL,
  TASKS_CLOSE_MODAL, TASKS_CHANGE_ORDER,
} from '../actionTypes/tasksActionTypes';

export const initialState = {
  tasks: [
    {
      id: 0,
      title: 'Example task',
      description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.',
      colors: {
        text: 'pink',
      },
    },
    {
      id: 1,
      title: 'Example task 2',
      description: '',
      colors: {
        text: 'purple',
      },
    },
    {
      id: 2,
      title: 'Example task 3',
      description: '',
      colors: {
        text: 'yellow',
      },
    },
  ],
  nextId: 3,
  isEditModalOpen: false,
  currentEditTaskId: null,
  isAddModalOpen: false,
};

const tasksReducer = (state = initialState, action) => {
  switch (action && action.type) {
    case TASKS_ADD: {
      const newTodo = {
        ...action.task,
        id: state.nextId,
      };

      return {
        ...state,
        nextId: state.nextId + 1,
        tasks: [...state.tasks, newTodo],
      };
    }
    case TASKS_REMOVE: {
      return {
        ...state,
        tasks: state.tasks.filter((task) => task.id !== action.id),
      };
    }
    case TASKS_COMPLETE:
      return {
        ...state,
        tasks: state.tasks.filter((task) => task.id !== action.id),
      };
    case TASKS_OPEN_ADD_MODAL:
      return {
        ...state,
        isAddModalOpen: true,

      };
    case TASKS_CLOSE_MODAL:
      return {
        ...state,
        isEditModalOpen: false,
        isAddModalOpen: false,
      };
    case TASKS_OPEN_EDIT_MODAL:
      return {
        ...state,
        isEditModalOpen: true,
        currentEditTaskId: action.id,
      };
    case TASKS_EDIT: {
      const { currentEditTaskId } = state;
      return {
        ...state,
        tasks: [
          ...state.tasks.filter((task) => task.id !== action.id),
          { ...action.editedTask, id: currentEditTaskId },
        ],
        currentEditTaskId: null,
      };
    }
    case TASKS_CHANGE_ORDER: {
      const {orderDown, id} = action;
      const tasksCopy = [...state.tasks];
      let tasksToOrderIndex = 0;

      for( ; tasksToOrderIndex < tasksCopy.length; ++tasksToOrderIndex){
        if(tasksCopy[tasksToOrderIndex].id === id){
          break;
        }
      }

      if(orderDown){
        const previousId = tasksCopy[tasksToOrderIndex+1].id;
        tasksCopy[tasksToOrderIndex].id = previousId;
        tasksCopy[tasksToOrderIndex+1].id = id;
      } else {
        const previousId = tasksCopy[tasksToOrderIndex-1].id;
        tasksCopy[tasksToOrderIndex].id = previousId;
        tasksCopy[tasksToOrderIndex-1].id = id;
      }

      return {
        ...state,
        tasks: tasksCopy
      };
    }
    default: return state;
  }
};

export default tasksReducer;
