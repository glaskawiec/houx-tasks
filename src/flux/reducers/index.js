import tasksReducer from './tasksReducer';

const reducers = {
  tasks: tasksReducer,
};

export default reducers;
