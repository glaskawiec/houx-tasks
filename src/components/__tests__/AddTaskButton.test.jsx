import { render } from '@testing-library/react';
import React from 'react';
import { HouxProvider } from 'houx';
import reducers from '../../flux/reducers';
import AddTaskButton from '../AddTaskButton';

describe('AddTaskButton component unit tests', () => {
  test('AddTaskButton should render properly', () => {
    const { container } = render(
      <HouxProvider reducers={reducers}>
        <AddTaskButton />
      </HouxProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
