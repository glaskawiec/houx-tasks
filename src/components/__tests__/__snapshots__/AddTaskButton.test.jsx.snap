// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTaskButton component unit tests AddTaskButton should render properly 1`] = `
<div>
  <button
    class="btn btn-primary"
    data-testid="add-task-button"
    type="button"
  >
    Add Task
  </button>
</div>
`;
